examples/ColorCamera/rgb_undistort.py

```diff
@@ -0,0 +1,109 @@
+import time
+import cv2
+import depthai as dai
+import numpy as np
+
+resolutionMap = {"THE_1080_P": (1920, 1080), "THE_800_P": (1280, 800), "THE_720_P": (1280, 720), "THE_400_P": (640, 400)}
+resolution = resolutionMap["THE_1080_P"]
+camSocket = dai.CameraBoardSocket.RGB
+meshWidth = 2
+meshHeight = 2
+
+def getMesh(calibData):
+    global meshWidth
+    global meshHeight
+
+    M1 = np.array(calibData.getCameraIntrinsics(camSocket, resolution[0], resolution[1]))
+    d1 = np.array(calibData.getDistortionCoefficients(camSocket))
+    R1 = np.identity(3)
+    mapX, mapY = cv2.initUndistortRectifyMap(M1, d1, R1, M1, resolution, cv2.CV_32FC1)
+
+    meshCellSize = 16
+    mesh0 = []
+    # Creates subsampled mesh which will be loaded on to device to undistort the image
+    for y in range(mapX.shape[0] + 1): # iterating over height of the image
+        if y % meshCellSize == 0:
+            rowLeft = []
+            for x in range(mapX.shape[1]): # iterating over width of the image
+                if x % meshCellSize == 0:
+                    if y == mapX.shape[0] and x == mapX.shape[1]:
+                        rowLeft.append(mapX[y - 1, x - 1])
+                        rowLeft.append(mapY[y - 1, x - 1])
+                    elif y == mapX.shape[0]:
+                        rowLeft.append(mapX[y - 1, x])
+                        rowLeft.append(mapY[y - 1, x])
+                    elif x == mapX.shape[1]:
+                        rowLeft.append(mapX[y, x - 1])
+                        rowLeft.append(mapY[y, x - 1])
+                    else:
+                        rowLeft.append(mapX[y, x])
+                        rowLeft.append(mapY[y, x])
+            if (mapX.shape[1] % meshCellSize) % 2 != 0:
+                rowLeft.append(0)
+                rowLeft.append(0)
+
+            mesh0.append(rowLeft)
+
+    mesh0 = np.array(mesh0)
+    meshWidth = mesh0.shape[1] // 2
+    meshHeight = mesh0.shape[0]
+    mesh0.resize(meshWidth * meshHeight, 2)
+
+    mesh = list(map(tuple, mesh0))
+
+    return mesh
+
+def create_pipeline(mesh):
+    pipeline = dai.Pipeline()
+
+    cam = pipeline.create(dai.node.ColorCamera)
+    cam.setBoardSocket(camSocket)
+    cam.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+    cam.setIspScale(1, 1)
+
+    manip1 = pipeline.create(dai.node.ImageManip)
+    manip1.setWarpMesh(mesh, meshWidth, meshHeight)
+    manip1.setMaxOutputFrameSize(cam.getIspWidth() * cam.getIspHeight() * 3 // 2)
+    # TODO manip1.initialConfig.setWarpBorderFillColor(0, 0, 0)
```

Comment:
test if it works, if not then remove.
Outliers should be set to 0 (black) ideally,

Reply:
same comment